connectors/vault/vault_connector.go

```diff
@@ -5,52 +5,56 @@ package main
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"log"
 	"net"
 	"strconv"
-	"strings"
 
+	. "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	vaultbl "github.com/ibm/the-mesh-for-data/connectors/vault/vaultconn-bl"
 	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
 	"google.golang.org/grpc"
 )
 
-var vault VaultConnection
-
 const defaultTimeout = "180"
 const defaultPort = "50083" //synched with vault_connector.yaml
 
 type server struct {
 	pb.UnimplementedDataCredentialServiceServer
+	vaultConnector *vaultbl.Server
 }
 
-func (s *server) GetCredentialsInfo(ctx context.Context, in *pb.DatasetCredentialsRequest) (*pb.DatasetCredentials, error) {
-	log.Println("Vault connector: GetCredentialsInfo")
-	log.Printf("Vault connector: GetCredentialsInfo, in = %v\n", in)
+var vault VaultConnection
 
-	// secretaddr := url.QueryEscape(in.DatasetId)
-	secretaddr := strings.ReplaceAll(in.DatasetId, " ", "")
+func (s *server) GetCredentialsInfo(ctx context.Context, in *pb.DatasetCredentialsRequest) (*pb.DatasetCredentials, error) {
+	log.Println("Received ApplicationContext")
+	log.Println(in)
 
-	readCredentials, err := vault.GetFromVault(secretaddr)
+	vaultPathKey := GetEnv(VaultPathKey)
+	eval, err := s.vaultConnector.GetCredentialsInfo(in, vault, vaultPathKey)
 	if err != nil {
 		log.Printf("Error in vaultConnector, got error from the vault: %v\n", err.Error())
-		return nil, fmt.Errorf("error in retrieving the secret from vault(key = %s): %v", secretaddr, err)
+		return nil, fmt.Errorf("error in retrieving the secret from vault: %v", err)
 	}
-	log.Println("Read credentials from vault: " + readCredentials)
+	jsonOutput, _ := json.MarshalIndent(eval, "", "\t")
+	log.Println("Received evaluation : " + string(jsonOutput))
+	log.Println("err:", err)
```

Comment:
Why do you log the error here? If it's null there is no point in logging. If there is an error you already return early.

Reply:
fixed. Thanks.